Upstream: yes

From 6a4dbb07ccf92836bb4adac7be4575672d0ac5f1 Mon Sep 17 00:00:00 2001
From: erouault <erouault>
Date: Mon, 15 Aug 2016 20:49:48 +0000
Subject: [PATCH] * libtiff/tif_pixarlog.c: Fix write buffer overflow in
 PixarLogEncode if more input samples are provided than expected by
 PixarLogSetupEncode. Idea based on libtiff-CVE-2016-3990.patch from
 libtiff-4.0.3-25.el7_2.src.rpm by Nikola Forro, but with different and
 simpler check. (bugzilla #2544)

invalid tests that rejected valid files. (bugzilla #2545)
---
 libtiff/tif_pixarlog.c |  7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/libtiff/tif_pixarlog.c b/libtiff/tif_pixarlog.c
index e78f788..28329d1 100644
--- a/libtiff/tif_pixarlog.c
+++ b/libtiff/tif_pixarlog.c
@@ -1141,6 +1141,13 @@ PixarLogEncode(TIFF* tif, uint8* bp, tmsize_t cc, uint16 s)
 	}
 
 	llen = sp->stride * td->td_imagewidth;
+    /* Check against the number of elements (of size uint16) of sp->tbuf */
+    if( n > td->td_rowsperstrip * llen )
+    {
+        TIFFErrorExt(tif->tif_clientdata, module,
+                     "Too many input bytes provided");
+        return 0;
+    }
 
 	for (i = 0, up = sp->tbuf; i < n; i += llen, up += llen) {
 		switch (sp->user_datafmt)  {
-- 
2.9.3

